### PR TITLE
general-concepts/manifest: discuss concerns with regenerating existin…

### PR DESCRIPTION
**Pull request by thesamesam**
…g distfile entries

In a discussion with Eli Schwartz, I realised the devmanual doesn't currently explicitly state best practice.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
Thank you, much appreciated.

File: general-concepts/manifest/text.xml [hunk 1]
```diff
@@ -39,6 +39,42 @@ action from developers.
...
+Developers should diff the old and new versions of the distfile, comparing
+the two, and note the differences in the commit message updating the Manifest
+to indicate both what happened (if any context is known) and also what differences
+between the two distfiles have been ascertained.
+</p>
+
+<p>
+Please note that if upstream made any changes affecting the built
+package or it had substantial differences, you need to also bump the ebuild's revision.
+Finally,remember to remove the ebuilds that are associated with the old
```
Comment (ulm) [inline general-concepts/manifest/text.xml:63]:
```suggestion
Finally, remember to remove the ebuilds that are associated with the old
```

File: general-concepts/manifest/text.xml [hunk 2]
```diff
@@ -39,6 +39,42 @@ action from developers.
...
+distfile, or regenerate their checksums in <c>Manifest</c>, if there
+are any. This is necessary because these ebuilds will cause checksum
+mismatch errors as the checksum recorded in the Manifest file no
+longer matches the computed checksum of the fetched distfile.
+</p>
+
+<p>
+Special care is also required with regard to mirrors, with
+<uri link="::general-concepts/mirrors#replacing-automatically-mirrored-files">
```
Comment (ulm) [inline general-concepts/manifest/text.xml:72]: For the link target, please use the section name from the XML source, not the label from the generated HTML:
```suggestion
<uri link="::general-concepts/mirrors#Replacing Automatically Mirrored Files">
```

File: general-concepts/manifest/text.xml [hunk 3]
```diff
@@ -39,6 +39,42 @@ action from developers.
 </p>
 </body>
 </subsection>
+
+<subsection>
+<title>Updating Manifests</title>
```
Comment (ulm) [inline general-concepts/manifest/text.xml:44]:
```suggestion
<title>Updating Manifest files</title>
```

File: general-concepts/manifest/text.xml [hunk 4]
```diff
@@ -39,6 +39,42 @@ action from developers.
...
+<body>
+<p>
+Updating existing entries within a Manifest must be done with care. Upstream
```
Comment (ulm) [inline general-concepts/manifest/text.xml:47]: I think that throughout this should either be spelled `manifest` in lower case if you refer to the concept, or `the <c>Manifest</c> file` if you refer to the actual file.

```suggestion
Updating existing entries within a <c>Manifest</c> file must be done with care. Upstream
```

File: general-concepts/manifest/text.xml [hunk 5]
```diff
@@ -39,6 +39,42 @@ action from developers.
...
+changing the tarball in-place without a new filename could be an innocent
+respin of the tarball, or it could indicate either the previous or the new
+tarball is malicious.
+</p>
+
+<p>
+Developers should diff the old and new versions of the distfile, comparing
+the two, and note the differences in the commit message updating the Manifest
```
Comment (ulm) [inline general-concepts/manifest/text.xml:55]:
```suggestion
the two, and note the differences in the commit message updating the <c>Manifest</c> file
```

File: general-concepts/mirrors/text.xml [hunk 6]
```diff
@@ -68,13 +68,8 @@ fetch and start distributing the new version.
...
 <p>
-Please note that if upstream made any changes affecting the built
-package, you need to also bump the ebuild's revision. Finally,
-remember to remove the ebuilds that are associated with the old
-distfile, or regenerate their checksums in <c>Manifest</c>, if there
-are any. This is necessary because these ebuilds will cause checksum
-mismatch errors as the checksum recorded in the Manifest file no
-longer matches the computed checksum of the fetched distfile.
+Updating an existing distfile is generally cause for concern and must be done
+with care, see <uri link="::general-concepts/manifest#updating_manifests"/>.
```
Comment (ulm) [inline general-concepts/mirrors/text.xml:72]:
```suggestion
with care, see <uri link="::general-concepts/manifest#Updating Manifests"/>.
```

**Review by ulm — COMMENTED** [review]
Thanks, this is a good addition.

Please follow the [style guide](https://devmanual.gentoo.org/appendices/devbook-guide/index.html#internal-coding-style):
- Empty line before opening `<p>` tags
- Word wrapping at 80 characters

File: general-concepts/manifest/text.xml [hunk 7]
```diff
@@ -39,6 +39,42 @@ action from developers.
...
+package or it had substantial differences, you need to also bump the ebuild's revision.
+Finally,remember to remove the ebuilds that are associated with the old
+distfile, or regenerate their checksums in <c>Manifest</c>, if there
+are any. This is necessary because these ebuilds will cause checksum
+mismatch errors as the checksum recorded in the Manifest file no
+longer matches the computed checksum of the fetched distfile.
+</p>
+
+<p>
+Special care is also required with regard to mirrors, with
```
Comment (sbraz) [inline general-concepts/manifest/text.xml:71]: "with regard to mirrors" is repeated here.

**Comment by thesamesam** [comment]
> Thanks, this is a good addition.
> 
> Please follow the [style guide](https://devmanual.gentoo.org/appendices/devbook-guide/index.html#internal-coding-style):
> 
>     * Empty line before opening `<p>` tags
> 
>     * Word wrapping at 80 characters

Thanks, sorry, I've been out of the devmanual for too long. Need to get back into it..

**Comment by thesamesam** [comment]
I think I need you to point out where the `<p` rule was violated. My use appears consistent with manifest/text.xml and mirrors/test.xml. Do you want a new line before the first `<p` in the file too? The other uses don't seem to do that.

**Comment by ulm** [comment]
> I think I need you to point out where the `<p` rule was violated. My use appears consistent with manifest/text.xml and mirrors/test.xml. Do you want a new line before the first `<p` in the file too? The other uses don't seem to do that.

Line 46.

And yes, there may be some places where the style guide wasn't followed, but let's follow it for new additions.

**Comment by thesamesam** [comment]
No worries, I wasn't arguing about the style, more that I was worried I was misunderstanding it.

File: general-concepts/manifest/text.xml [hunk 8]
```diff
@@ -39,6 +39,43 @@ action from developers.
 </p>
 </body>
 </subsection>
+
+<subsection>
+<title>Updating Manifest files</title>
```
Comment (ulm) [inline general-concepts/manifest/text.xml:44]:
```suggestion
<title>Updating <c>Manifest</c> files</title>
```

File: general-concepts/manifest/text.xml [hunk 9]
```diff
@@ -39,6 +39,43 @@ action from developers.
...
+also what differences between the two distfiles have been ascertained.
+</p>
+
+<p>
+Please note that if upstream made any changes affecting the built
+package or it had substantial differences, you need to also bump the ebuild's
+revision. Finally, remember to remove the ebuilds that are associated with the
+old distfile, or regenerate their checksums in <c>Manifest</c>, if there
+are any. This is necessary because these ebuilds will cause checksum
+mismatch errors as the checksum recorded in the Manifest file no
```
Comment (ulm) [inline general-concepts/manifest/text.xml:67]: Either:
```suggestion
mismatch errors as the checksum recorded in the <c>Manifest</c> file no
```
or:
```suggestion
mismatch errors as the checksum recorded in the manifest file no
```

File: general-concepts/manifest/text.xml [hunk 10]
```diff
@@ -39,6 +39,43 @@ action from developers.
...
+longer matches the computed checksum of the fetched distfile.
+</p>
+
+<p>
+Special care is also required with
+<uri link="::general-concepts/mirrors/#Replacing Automatically Mirrored Files">
+regard to mirrors.
+</uri>
```
Comment (ulm) [inline general-concepts/manifest/text.xml:75]: I suggest using just "mirrors" as the link text (and move the full stop out of it):
```suggestion
Special care is also required with regard to
<uri link="::general-concepts/mirrors/#Replacing Automatically Mirrored Files">
mirrors</uri>.
```

**Review by ulm — APPROVED** [review]

**Pull request closed without merging**